data-modelling.md

```diff
@@ -1 +1,129 @@
 # Data modelling
+
+The intention is to separate the NPQ application from ECF so that NPQ can operate standalone.
+
+In order to do that we need to move NPQ data from the ECF app to the NPQ one. This document
+will propose a model for storing that data.
+
+The aim is:
+
+* not to introduce anything unnecessary to NPQ
+* to structure the data so it's easy to understand
+* the queries that power the API should be simple and fast
+
+## Proposed schema
+
+_This is an initial version and likely to change_.
+
+```mermaid
+erDiagram
+    Application }|--|| User : ""
+    Application }|--|| LeadProvider : ""
+    Application }|--|| Course : ""
+    Application }o--o| School : ""
+    Application }o--o| PrivateChildcareProvider : ""
+    Application }|--|| Cohort : ""
+    %%Application }|--|| ParticipantIdentity : belongs_to
+
+    %%ParticipantIdentity }|--|| User : belongs_to
+
+    Declaration }|--|| User : ""
+    Declaration }|--|| Course : ""
+    Declaration }|--|| LeadProvider : ""
+    Declaration }|--|| Application : ""
+
+    Statement }|--|| LeadProvider : ""
+    Statement }|--|| Cohort : ""
+
+    StatementLineItem }|--|| Statement : ""
+    StatementLineItem }|--|| Declaration : ""
+
+    Contract }|--|| LeadProvider : ""
+    Contract }|--|| Cohort : ""
+    Contract }|--|| Course : ""
+    %% Contract linked to Statement with version
+    Contract }|--|| Statement : ""
+
+    Schedule }|--|| Cohort : ""
+    
+    Milestone }|--|| Schedule : ""
+
+    ParticipantOutcome }|--|| Declaration : ""
+
+    User {
+        uuid id
+        string email
+        string full_name
+        string teacher_reference_number
+        datetime updated_at
+    }
+
+    LeadProvider {
+        uuid id
+        string name
+    }
+
+    Course {
+        uuid id
+        string identifier "Is this unique?"
+    }
+
+    Application {
+        uuid id
+        uuid course_id
+        uuid lead_provider_id
+        uuid participant_identity_id
+        string employer_name
+        string employment_role
+        string funding_choice
+        string headteacher_status
+        string ineligible_for_funding_reason
+        string private_childcare_provider_urn
+        string teacher_reference_number
+        boolean teacher_reference_number_verified
+        uuid school_id
```

Comment:
I think we should just use the URN as the key for schools.

Reply:
If we use URN here, we should look to use URN as primary key for `School` table